src/components/SearchModal/CurrencySearch.tsx

```diff
@@ -325,7 +333,35 @@ export function CurrencySearch({
       const nextPage = (page ?? pageCount) + 1
       let tokens: WrappedTokenInfo[] = []
       if (debouncedQuery) {
-        tokens = await fetchTokens(debouncedQuery, nextPage, chainId)
+        const promiseResult = await Promise.allSettled([
+          fetchTokens(debouncedQuery, nextPage, chainId),
+          fetchTokenFromRPC(debouncedQuery),
+        ])
+
+        const fetchTokensResp = promiseResult[0]
+        const fetchTokenFromRPCResp = promiseResult[1]
```

Comment:
should check if `debouncedQuery` is an address, only call if it's an valid evm address

Reply:
Sure